src/classes/issues-processor.ts

```diff
@@ -122,308 +123,313 @@ export class IssuesProcessor {
 
     for (const issue of issues.values()) {
       const issueLogger: IssueLogger = new IssueLogger(issue);
-      this._statistics?.incrementProcessedItemsCount(issue);
+      await issueLogger.grouping(`$$type #${issue.number}`, async () => {
```

Comment:
can you just pass in the `this.processIssue(issue, actor)` instead of wrapping it in another async call? Or does that make Typescript unhappy?

Reply:
Unfortunately, that is not possible as far as I know. We can just push the async call further into the methods.

Reply:
It's hardcore to review.
I am not convinced that we really need to have async grouping.
I think it's only useful for parallel processing which is not the case here.
Can we try a basic version and see if the group method without synchronous calls work?

Reply:
I have tried to use the method `startGroup` and `endGroup` so I can remove the `async` for the `grouping` method. But the problem is that the method in the `gouping` method must be `async` because the method `processIssue` is `async`. With this setup I tested it but I got only failed tests and it does not work for me like expected. Maybe you have more success?!

Reply:
I think the async is fine in this case. it's clean